frontend/apps/marketing/src/components/link/Link.tsx

```diff
@@ -0,0 +1,52 @@
+import {MarginBottomNoneToM} from '@/components/common/types';
+import '@code-dot-org/component-library/link/index.css';
+import {default as DSCOLink} from '@code-dot-org/component-library/link';
+import {ComponentSizeXSToL} from '@code-dot-org/component-library/common/types';
+import FontAwesomeV6Icon from '@code-dot-org/component-library/fontAwesomeV6Icon';
+import React, {ReactNode} from 'react';
+import classNames from 'classnames';
+
+import moduleStyles from './link.module.scss';
+
+export type LinkProps = {
+  /** Link content */
+  children: ReactNode;
+  /** Link URL */
+  href: string;
+  /** Link color */
+  color: 'purple' | 'white';
+  /** Link margin bottom */
+  marginBottom: MarginBottomNoneToM;
+  /** Link size */
+  size: ComponentSizeXSToL;
+  /** Whether Link is for internal code.org pages, or external web page. (external links are opened in new tab) */
+  isLinkExternal: boolean;
```

Comment:
Do we have a task created to automate this logic? Would be good to add a jira to locations where this exists so we know all the updates that are needed.

Reply:
We don't have a separate ticket for this yet, if you have time for this - please feel free to do or code this. For now I keep everything in my head so feel free to ask. (Basically right now it's used only in 2 places - Link and Button components, and it looks like at least in this sprint there won't be any additional usages of isLinkExternal prop or anything like that). Also come discussions are saved here - https://github.com/code-dot-org/code-dot-org/pull/63558.  Last comment - https://github.com/code-dot-org/code-dot-org/pull/63558#discussion_r1946887560

This PR blocks other higher priority tickets that I have in this sprint, so I hope this PR review to be fast